Quick win: Cancel tracker animations on click

0nko opened the pull request:
Task/Issue URL: https://app.asana.com/1/137249556945/project/1207418217763355/task/1212330168162873

### Description

Adds animation canceling when clicked to allow users to start using the browser more quickly.

### Steps to test this PR

_Regular tab_
- [x] Load a website with trackers
- [x] While animation is in progress, tap on it
- [x] Verify the animation is canceled and you can start typing in the address bar

_Custom tab_
- [x] Go to Settings -> Developer settings -> Custom tabs)
- [x] Load a website with trackers
- [x] While animation is in progress, tap on it
- [x] Verify the animation is canceled


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> **Low Risk**
> Low risk UI/animation behavior changes limited to canceling and resetting omnibar/custom tab tracker animations; main risk is introducing visual glitches or leaving views in an incorrect alpha/state if cancellation paths are missed.
> 
> **Overview**
> Allows users to interrupt tracker-related address bar animations by tapping, so the omnibar/custom tab toolbar becomes usable immediately.
> 
> Adds click-to-cancel hooks for the trackers animation views (regular tabs and custom tabs), expands `cancelAnimations` to also restore shield view visibility, and hardens cancellation logic to prevent late callbacks from restarting animations (e.g., guard `playAnimation`, flip `isAnimationRunning` earlier, and explicitly cancel/clear pending cookie animators/transitions).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4e4d1b74dde71d4166aff32fc0c1fb9884bf9fda. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

0nko committed:
Cancel custom tab and omnibar animations when they are clicked

0nko committed:
Revert the URL copying

0nko committed:
Fix the shield being invisible after canceling cookie animation

0nko committed:
Fix the cookie animation cancelation

mikescamell commented:
I think this will need more work, and might be more than a QWD attempt 😬 

- Let cookies play
- Cancel it
- Refresh
- You get a glimpse of the cookies animation before the trackers animation plays

https://github.com/user-attachments/assets/8ee6b651-e457-42fd-b669-1d72605dcbe9

0nko committed:
Fix the cookie animation glitch

mikescamell committed:
Prevent playing animation if animation has been cancelled

You can cancel the animation just as the shield pop is about to happen. even if it does not play, when you load the animation again we end up in a state where the shield pop will play first before the trackers count animation

0nko committed:
Prevent the animator callback from being called when animation is cancelled

0nko committed:
Fix the build

0nko committed:
Make sure the cleanup always happens

GerardPaligot commented on app/src/main/java/com/duckduckgo/app/browser/omnibar/animations/addressbar/BrowserLottieTrackersAnimatorHelper.kt:
```diff
@@ -408,13 +427,31 @@ class BrowserLottieTrackersAnimatorHelper @Inject constructor(
     private fun stopCookiesAnimation() {
         if (!::cookieViewBackground.isInitialized || !::cookieView.isInitialized) return
 
+        // Set flag BEFORE canceling to prevent callbacks from continuing the old animation
```
Suggestion: Add a condition to avoid to execute the following function if it isn't necessary:

```
// Early return if no animations are running to avoid redundant work
if (hasCookiesAnimationBeenCanceled && runningCookieAnimators.isEmpty()) return
```

GerardPaligot reviewed (commented):
I'm still having a strange behavior:

* At 1 minute, I'm opening "decathlon.fr" website,
* a cookie popup dialog is blocked with a quick animation in the omnibar,
* then trackers are blocked with a longer animation in the omnibar,
* few seconds later, cookie animation is played again in the omnibar

https://github.com/user-attachments/assets/e30ebd30-1378-493f-869a-74e8396eda0e
